packages/@aws-cdk/aws-servicecatalog/lib/private/validation.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Class to validate that inputs match requirements.
+ */
+export class InputValidator {
+  /**
+   * Validates length is between allowed min and max lengths.
+   */
+  public static validateLength(resourceName: string, inputName: string, minLength: number, maxLength: number, inputString?: string): void {
+    if (inputString !== undefined && (inputString.length < minLength || inputString.length > maxLength)) {
+      throw new Error(`Invalid ${inputName} length of ${inputString.length} for ${resourceName}, ${inputName} length must be between ${minLength} and ${maxLength}\n` +
```

Comment:
This is way, way overdone I think 🙂.

```suggestion
      throw new Error(`Invalid ${inputName}, must be between ${minLength} and ${maxLength}, got: '${inputString}'`);
```

Reply:
you mentioned before though that we wanted to print the value causing the error as well, this would remove that, is that okay now?

Reply:
I still want to truncate this, if the input was somehow a description thousands of characters long, it would be an annoying error message to output the whole thing